Applications/deviceconfiguration/simpleconfig.py

```diff
@@ -40,14 +40,75 @@
 ###Update configuration using INI file as defined by Faraday device object and functions
 #########################################################################################
 
+# Send POST data to Proxy to configure unit
 try:
-    print "Sending configuration to {0}-{1}".format(local_device_callsign,local_device_node_id)
     r = requests.post('http://{0}:{1}'.format(hostname,port), params={'callsign': str(local_device_callsign), 'nodeid': int(local_device_node_id)})
-    print "Completed configuration to {0}-{1}".format(local_device_callsign,local_device_node_id)
 
-except requests.exceptions.RequestException as e:  # This is the correct syntax
-    print e
+except requests.exceptions.RequestException as e:
+    # Some error occurred
+    print r.text
 
+#Check to see if programming was successful (HTTP 204 response)
+if r.status_code != 204:
+    print r.text
+else:
+    # Programming apparently successful. Let unit reboot and then query for flash data
+    
+    timer = 5 #  Wait five seconds
+    print "Programmed Faraday, waiting {0} seconds for reboot".format(str(timer))
+    while(timer > 0):
+        time.sleep(1) # Sleep to allow unit to process, polling and slow, not sure why THIS slow...
+        timer += -1
+
+    try:
+        r = requests.get("http://127.0.0.1:8002", params={'callsign': str(local_device_callsign), 'nodeid': int(local_device_node_id)})
+    except requests.exceptions.RequestException as e:
+        # Some error occurred
+        print r.text
+
+    # Obtain JSON response with data from Faraday
+    raw_unit_json = r.json()
+
+    # Decode and depickle (serialize) device configuration parsed dictionary data
+    b64_unit_json = base64.b64decode(raw_unit_json['data'])
+    unit_configuration_dict = cPickle.loads(b64_unit_json)
+    
+    #Print configuration values
+    print "\n************************************"
+    
+    print "POST CONFIGURATION UPDATE"
+    print "\nBASIC"
+    print "Unit Callsign:", str(unit_configuration_dict['local_callsign'])[0:unit_configuration_dict['local_callsign_length']]
+    print "ID:", str(unit_configuration_dict['local_callsign_id'])
+    print "CONFIGBOOTBITMASK:", bin(unit_configuration_dict['configuration_bitmask'])
```

Comment:
Since this is a binary print of the bitmask you should show all leading zeros: http://stackoverflow.com/questions/16926130/python-convert-to-binary-and-keep-leading-zeros